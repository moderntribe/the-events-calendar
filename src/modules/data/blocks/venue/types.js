/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_VENUE = `${ PREFIX_EVENTS_STORE }/SET_VENUE`;
export const SET_VENUE_MAP = `${ PREFIX_EVENTS_STORE }/SET_VENUE_MAP`;
export const SET_VENUE_MAP_LINK = `${ PREFIX_EVENTS_STORE }/SET_VENUE_MAP_LINK`;
