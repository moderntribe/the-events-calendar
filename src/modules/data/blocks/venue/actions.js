/**
 * Internal dependencies
 */
import * as types from './types';

export const setVenue = ( id ) => ( {
	type: types.SET_VENUE,
	payload: {
		venue: id,
	},
} );

export const removeVenue = () => ( {
	type: types.SET_VENUE,
	payload: {
		venue: 0,
	},
} );

export const setShowMap = ( showMap ) => ( {
	type: types.SET_VENUE_MAP,
	payload: {
		showMap,
	},
} );

export const setShowMapLink = ( showMapLink ) => ( {
	type: types.SET_VENUE_MAP_LINK,
	payload: {
		showMapLink,
	},
} );
