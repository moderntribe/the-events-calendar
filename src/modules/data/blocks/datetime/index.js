/**
 * Internal dependencies
 */
import reducer, { setInitialState, defaultStateToMetaMap } from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;
export { types, actions, selectors, sagas, setInitialState, defaultStateToMetaMap };
