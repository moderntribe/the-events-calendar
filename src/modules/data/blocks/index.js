/**
 * Internal dependencies
 */
import * as datetime from './datetime';
import subscribe from './subscribers';

export { datetime, subscribe };
export { default, setInitialState } from './reducer';
