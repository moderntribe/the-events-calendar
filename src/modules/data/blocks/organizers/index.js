/**
 * Internal dependencies
 */
import reducer, { setInitialState } from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import subscribe from './subscribers';

export default reducer;
export { types, actions, selectors, setInitialState, subscribe };
