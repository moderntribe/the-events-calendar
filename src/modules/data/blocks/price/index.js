/**
 * Internal dependencies
 */
import reducer, { setInitialState, defaultStateToMetaMap } from './reducer';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';
import * as utils from './utils';

export default reducer;
export { selectors, actions, types, utils, setInitialState, defaultStateToMetaMap };
