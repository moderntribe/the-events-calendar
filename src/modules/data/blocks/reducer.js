/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import datetime, { setInitialState as datetimeSetInitialState } from './datetime';
import organizers, { setInitialState as organizersSetInitialState } from './organizers';
import price, { setInitialState as priceSetInitialState } from './price';
import website, { setInitialState as websiteSetInitialState } from './website';
import venue, { setInitialState as venueSetInitialState } from './venue';

export const setInitialState = ( data ) => {
	datetimeSetInitialState( data );
	organizersSetInitialState( data );
	priceSetInitialState( data );
	venueSetInitialState( data );
	websiteSetInitialState( data );
};

export default combineReducers( {
	datetime,
	venue,
	organizers,
	price,
	website,
} );
