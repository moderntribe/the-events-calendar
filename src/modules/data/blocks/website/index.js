/**
 * Internal dependencies
 */
import reducer, { setInitialState, defaultStateToMetaMap } from './reducer';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';

export default reducer;
export { selectors, actions, types, setInitialState, defaultStateToMetaMap };
