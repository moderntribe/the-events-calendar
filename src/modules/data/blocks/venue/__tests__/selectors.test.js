/**
 * Internal dependencies
 */
import { DEFAULT_STATE } from '@moderntribe/events/data/blocks/venue/reducer';
import { selectors } from '@moderntribe/events/data/blocks/venue';

const state = {
	events: {
		blocks: {
			venue: DEFAULT_STATE,
		},
	},
};

jest.mock( '@moderntribe/common/utils/globals', () => ( {
	editorDefaults: () => ( {
		venue: 0,
		venueCountry: '',
		venueState: '',
		venueProvince: '',
	} ),
	list: () => ( {
		countries: {},
		us_states: {},
	} ),
	mapsAPI: () => ( {
		embed: true,
	} ),
} ) );

describe( '[STORE] - Venue selectors', () => {
	it( 'Should return the map embed config', () => {
		expect( selectors.getMapEmbed() ).toMatchSnapshot();
	} );

	it( 'Should return the venue block', () => {
		expect( selectors.venueBlockSelector( state ) ).toMatchSnapshot();
	} );

	it( 'Should select the showMap', () => {
		expect( selectors.getshowMap( state ) ).toMatchSnapshot();
	} );

	it( 'Should select the showMapLink', () => {
		expect( selectors.getshowMapLink( state ) ).toMatchSnapshot();
	} );

	it( 'Should select the venue', () => {
		expect( selectors.getVenue( state ) ).toMatchSnapshot();
	} );
} );
