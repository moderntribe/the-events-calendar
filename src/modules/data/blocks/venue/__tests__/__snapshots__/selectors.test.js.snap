// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Venue selectors Should return the map embed config 1`] = `true`;

exports[`[STORE] - Venue selectors Should return the venue block 1`] = `
Object {
  "showMap": true,
  "showMapLink": true,
  "venue": 0,
}
`;

exports[`[STORE] - Venue selectors Should select the showMap 1`] = `true`;

exports[`[STORE] - Venue selectors Should select the showMapLink 1`] = `true`;

exports[`[STORE] - Venue selectors Should select the venue 1`] = `0`;
